packages/flame/lib/src/components/position_component.dart

```diff
@@ -23,19 +23,23 @@ import 'mixins/hitbox.dart';
 /// within this component's (width, height).
 abstract class PositionComponent extends BaseComponent {
   /// The position of this component on the screen (relative to the anchor).
-  Vector2 position;
+  final Vector2 _position;
+  Vector2 get position => _position;
+  set position(Vector2 position) => _position.setFrom(position);
```

Comment:
Just to avoid shadowing

```suggestion
  set position(Vector2 newPosition) => _position.setFrom(newPosition);
```